frame/statement/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+		<T as frame_system::Config>::RuntimeEvent: From<pallet::Event<T>>,
+		<T as frame_system::Config>::RuntimeEvent: TryInto<pallet::Event<T>>,
+		sp_statement_store::BlockHash: From<<T as frame_system::Config>::Hash>,
+	{
+		fn offchain_worker(now: BlockNumberFor<T>) {
+			log::trace!(target: LOG_TARGET, "Collecting statements at #{:?}", now);
+			Pallet::<T>::collect_statements();
+		}
+	}
+}
+
+impl<T: Config> Pallet<T>
+where
+	<T as frame_system::Config>::AccountId: From<sp_statement_store::AccountId>,
+	sp_statement_store::AccountId: From<<T as frame_system::Config>::AccountId>,
+	<T as frame_system::Config>::RuntimeEvent: From<pallet::Event<T>>,
+	<T as frame_system::Config>::RuntimeEvent: TryInto<pallet::Event<T>>,
+	sp_statement_store::BlockHash: From<<T as frame_system::Config>::Hash>,
+{
+	/// Validate a statement against current state. This is supposed to be called by the statement
+	/// store on the host side.
+	pub fn validate_statement(
+		_source: StatementSource,
+		mut statement: Statement,
+	) -> Result<ValidStatement, InvalidStatement> {
+		sp_io::init_tracing();
+		log::debug!(target: LOG_TARGET, "Validating statement {:?}", statement);
+		let account: T::AccountId = match statement.proof() {
+			Some(Proof::OnChain { who, block_hash, event_index }) => {
+				if frame_system::Pallet::<T>::parent_hash().as_ref() != block_hash.as_slice() {
+					log::debug!(target: LOG_TARGET, "Bad block hash.");
+					return Err(InvalidStatement::BadProof)
+				}
+				let account: T::AccountId = (*who).into();
+				match frame_system::Pallet::<T>::event_no_consensus(*event_index as usize) {
+					Some(e) => {
+						statement.strip_proof();
+						if let Ok(Event::NewStatement { account: a, statement: s }) = e.try_into() {
+							if a != account || s != statement {
+								log::debug!(target: LOG_TARGET, "Event data mismatch");
+								return Err(InvalidStatement::BadProof)
+							}
+						} else {
+							log::debug!(target: LOG_TARGET, "Event type mismatch");
+							return Err(InvalidStatement::BadProof)
+						}
+					},
+					_ => {
+						log::debug!(target: LOG_TARGET, "Bad event index");
+						return Err(InvalidStatement::BadProof)
+					},
+				}
+				account
+			},
+			_ => match statement.verify_signature() {
+				SignatureVerificationResult::Valid(account) => account.into(),
+				SignatureVerificationResult::Invalid => {
+					log::debug!(target: LOG_TARGET, "Bad statement signature.");
+					return Err(InvalidStatement::BadProof)
+				},
+				SignatureVerificationResult::NoSignature => {
+					log::debug!(target: LOG_TARGET, "Missing statement signature.");
+					return Err(InvalidStatement::NoProof)
+				},
+			},
+		};
+		let statement_cost = T::StatementCost::get();
+		let byte_cost = T::ByteCost::get();
+		let priority_cost = statement_cost;
+		let balance = <T::Currency as Inspect<AccountIdOf<T>>>::balance(&account);
+		let global_priority =
+			balance.checked_div(&priority_cost).unwrap_or_default().saturated_into();
```

Comment:
So this is the same priority for all statements of one account? (Assuming that the account balance doesn't change in between)

Reply:
This global priority is used to decide which statement from which account to evict when the global size limit is reached. With proper setup of pricing these limits should not be reached, so this is more of a last resort measure. That said it should still probably account for the local priority as well. I'll tweak it